raiden/transfer/node.py

```diff
@@ -110,6 +111,29 @@ def get_token_network_by_token_address(
     return token_network_state
 
 
+def get_token_network_by_address(
```

Comment:
> Shouldn't this be in views.py?

My initial idea was that views would return immutable objects, to avoid corrupting the internal state of the node. Because of this I avoid using views in the state machines. (I did, however, break this rule once or twice :P )

> TokenNetworkID and TokenNetworkAddress are in our eyes different symantics. Should this take TokenNetworkAddress as a parameter instead of ID?

True, but I don't want to undo the work that @konradkonrad is doing. If we want anything else then we should probably discuss it and make an ADR.

> Why did you choose to introduce this function and replace views.get_token_network_by_identifier? I see that your version should have an O(1) complexity but how many payment networks should we be operating on for this to be a problem?

Complexity is exactly the reason. I don't know how many networks there will be.